### PR TITLE
Update SdfGenerator to save link and sensor data to file

### PR DESCRIPTION
**Pull request by iche033**
# 🎉 New feature

depends on:
* https://github.com/ignitionrobotics/sdformat/pull/753
* https://github.com/ignitionrobotics/sdformat/pull/757

## Summary

Currently the SdfGenerator only write out models and link poses to SDF file. This PR updates the class to pull more data out from ECM and save them to file. This is needed for saving changes to SDF after models are edited by the model editor 

Specifically, these new SDF elements are now being updated when the world is saved:
* link `<inertial>` and `<wind_mode>` elements
* `<sensor>` element - currently supports only sensor types listed in https://github.com/ignitionrobotics/sdformat/pull/753

## Test it

1. Launch ign-gazebo with `sensors_demo.sdf` or `sensors` world
2. Go to the top left hamburger menu and select `Save world as...`
3. Set the output file by using the `Browse` button or manually input the output sdf file path.
4. Click `OK` to save
5. Open and saved SDF world file and verify content.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: src/SdfGenerator.cc [hunk 1]
```diff
@@ -421,16 +441,254 @@ namespace sdf_generator
...
+          updateSensorElement(sensorElem, _ecm, sensorEnt);
+        sensorElem = sensorElem->GetNextElement("sensor");
+      }
+    }
+
+    return true;
+  }
+
+  /////////////////////////////////////////////////
+  bool updateSensorElement(sdf::ElementPtr _elem,
```
Comment (nkoenig) [inline src/SdfGenerator.cc:559]: Would it be possible to use the `sdf::Sensor` class along with its `ToElement` function?

Comment (nkoenig) [reply src/SdfGenerator.cc:559]: I'm working on `ToElement` functions for the sdf Model, Link, Visual, and Collision classes. This should help reduce the code in ign-gazebo.

Comment (iche033) [reply src/SdfGenerator.cc:559]: > Would it be possible to use the sdf::Sensor class along with its ToElement function?

yeah it's should be using the `sdf::Sensor::ToElement()` functions below. The various `if` checks below are for getting the correct sensor component from the entity. I also override the name and pose based on latest values from the ECM - not sure if it's really needed.

Comment (nkoenig) [reply src/SdfGenerator.cc:559]: Thanks. I'm shelving my `ToElement` additions. It's too many changes given the time line.

**Review by nkoenig — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**